src/generate/register.rs

```diff
@@ -1421,6 +1427,38 @@ fn add_from_variants<'a>(
     }
 }
 
+fn add_reexport_variant_derived_name(
+    mod_items: &mut TokenStream,
+    evs: &EnumeratedValues,
+    base: &EnumPath,
+    desc: &str,
+) {
+    let base_name = util::replace_suffix(&base.name, "");
+    let derived_name = if let Some(ref name) = &evs.name {
+        if let Some(ref usage) = &evs.usage {
+            match usage {
+                Usage::Read => Some(name.to_owned() + "R"),
+                Usage::Write => Some(name.to_owned() + "W"),
+                Usage::ReadWrite => Some(name.to_owned()),
+            }
```

Comment:
This should also use `ident`.

@Emilgardis this is yet one question we need to discuss next meeting.

Now we use:
common RW: `enum_name`
only R: `enum_name`
only W: `enum_name`
different R+W: `enum_name` +  `enum_writer_name`

Should we simplify it to:
RW: `enum_name`
R: ` enum_reader_name`
W: `enum_writer_name`

Or support both notation somehow?

Reply:
I support to simplify it.
For the following reasons.
- Simpler rules are better.
- Supporting both notations would increase maintenance costs.
- Even if the functionality changes between reads and writes, the bit names would be different and can be distinguished.

Further My Opinion .
I think it is not good that these notations are added by 'svd2rust'.
Since these notations are added by 'svdtool patch', it would be better to add the names for re-export by 'svdtool patch'.
I have created pull request https://github.com/rust-embedded/svdtools/pull/205 for svdtools for this opinion.